src/fs/feature/git.rs

```diff
@@ -334,3 +334,46 @@ fn index_status(status: git2::Status) -> f::GitStatus {
         _                                                => f::GitStatus::NotModified,
     }
 }
+
+fn current_branch(repo: &git2::Repository) -> Option<String>{
+    let head = match repo.head() {
+        Ok(head) => Some(head),
+        Err(ref e) if e.code() == git2::ErrorCode::UnbornBranch || e.code() == git2::ErrorCode::NotFound => return None,
+        Err(e) => {
+            error!("Error looking up Git branch: {:?}", e);
+            return None
+        }
+    };
+
+    if let Some(h) = head{
+        if let Some(s) = h.shorthand(){
+            return Some(s.to_owned());
+        }
+    }
+
+    None
+}
+
+impl f::SubdirGitRepo{
+    pub fn from_path(dir : &Path) -> Self{
+
+        let path = &reorient(&dir);
+        let g = git2::Repository::open(path);
+        if let Ok(repo) = g{
+
+            let branch = current_branch(&repo);
+            match repo.statuses(None) {
+                Ok(es) => {
+                    if es.iter().filter(|s| s.status() != git2::Status::IGNORED).count() > 0{
```

Comment:
```suggestion
                    if es.iter().filter(|s| s.status() != git2::Status::IGNORED).any(|| true) {
```
Should be a bit faster on big repos with a lot of changes, since you only want to check if there is more than zero item.